### PR TITLE
ci: refactor deploy script

### PR DESCRIPTION
**Pull request by henryiii**
Following our guidelines.

@ast0815 Could you please add "scikit-hep" as a maintainer or owner on PyPI so we can set up a scikit-hep token to upload just this package instead of using a username, which is not as safe.

### CODE REVIEW

**Comment by ast0815** [comment]
I invited "scikit-hep". Let me know if it worked.

**Comment by henryiii** [comment]
It did, thanks! Okay with me generating a token and replacing the secrets with that?

**Comment by ast0815** [comment]
Sure. Go ahead.

File: .github/workflows/pythonpublish.yml [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
-        pip install setuptools setuptools_scm toml wheel twine
-    - name: Build and publish
-      env:
-        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
-        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-      run: |
-        python setup.py sdist bdist_wheel --universal
-        twine upload dist/*
+        user: ${{ secrets.PYPI_USERNAME }}
+        password: ${{ secrets.PYPI_PASSWORD }}
```
Comment (henryiii) [inline .github/workflows/pythonpublish.yml:43]:
```suggestion
        password: ${{ secrets.PYPI_PASSWORD }}
```

File: .github/workflows/pythonpublish.yml [hunk 2]
```diff
@@ -1,31 +1,43 @@
...
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release' && github.event.action == 'published'
+
+    steps:
+    - uses: actions/download-artifact@v2
+      with:
+        name: artifact
+        path: dist
+
+    - uses: pypa/gh-action-pypi-publish@v1.4.1
```
Comment (henryiii) [inline .github/workflows/pythonpublish.yml:40]:
```suggestion
    - uses: pypa/gh-action-pypi-publish@v1.4.2
```

**Comment by henryiii** [comment]
Done, change to "new" password (is the token) and removed username.

**Pull request merged**